### PR TITLE
[se-99] Setup authentication

### PR DESCRIPTION
**Pull request by ervin-khamoido**

### CODE REVIEW

File: src/utils/constants.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const APPOLO_CLIENT_URL = 'http://31.43.234.16:8087/graphql';
```
Comment (nikitalyashenko) [inline src/utils/constants.ts:1]: Please set it in ENV file

**Review by nikitalyashenko — CHANGES_REQUESTED** [review]
Nice work

File: src/providers/SessionProvider/SessionProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    hash: initData?.hash ?? '',
+    query_id: initData?.queryId ?? '',
+    user: {
+      id: initData?.user?.id,
+      first_name: initData?.user?.firstName,
+      last_name: initData?.user?.lastName,
+      username: initData?.user?.username || '',
+      language_code: initData?.user?.languageCode,
+      allows_write_to_pm: initData?.user?.allowsWriteToPm,
+    } as TelegramWebAppDataUserInput,
```
Comment (karpolan) [inline src/providers/SessionProvider/SessionProvider.tsx:37]: `const data: AccessTokenParams: TelegramWebAppDataUserInput = {`

File: src/providers/SessionProvider/SessionProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  };
+
+  useEffect(() => {
+    const handleRequest = async (webAppData: AccessTokenParams) => {
+      setIsLoading(true);
+
+      const accessToken = localStorage.getItem('access_token');
```
Comment (karpolan) [inline src/providers/SessionProvider/SessionProvider.tsx:44]: 'access_token' should be a const

File: src/providers/SessionProvider/SessionProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        .then(async (result) => {
+          if (result.data) {
+            setSessionToken(`Bearer ${result.data.telegramUserLogin.access_token}`);
+            localStorage.setItem('access_token', result.data.telegramUserLogin.access_token);
+          }
+        })
+        .catch(async (error) => {
+          setError(isGraphqlError(error, 'FULL_MAINTENANCE') ?? error.message);
+        })
+        .finally(() => setIsLoading(false));
```
Comment (karpolan) [inline src/providers/SessionProvider/SessionProvider.tsx:71]: WTF? Why not using Acync + Await?

File: src/utils/constants.ts (hunk 1 above)
Comment (karpolan) [reply src/utils/constants.ts:1]: And rename file to `config.ts`

**Review by karpolan — APPROVED** [review]
Try to use async await instead of Promises hell

File: src/providers/SessionProvider/SessionProvider.tsx (hunk 2 above)
Comment (ervin-khamoido) [reply src/providers/SessionProvider/SessionProvider.tsx:37]: fixed

File: src/providers/SessionProvider/SessionProvider.tsx (hunk 4 above)
Comment (ervin-khamoido) [reply src/providers/SessionProvider/SessionProvider.tsx:71]: fixed

**Pull request merged**